code-studio/src/js/components/StudioCellEditor.jsx

```diff
@@ -0,0 +1,109 @@
+    ['tileType', 'speed', 'size', 'direction', 'emotion', 'sprite'].forEach(value => {
+      if (values[value] === undefined) {
+        values[value] = 'undefined';
+      }
+    });
+
+    var spriteAttributes;
+    if (values.tileType === constants.SquareType.SPRITESTART) {
+      spriteAttributes = (
+        <div>
+          <label htmlFor="sprite">Sprite:</label>
+          <select name="sprite" value={values.sprite} onChange={this.handleChange}>
+            <option value="undefined">default</option>
+            {avatarList.map((sprite, i) => {
+              return <option key={sprite} value={i}>{sprite}</option>;
+            })}
+          </select>
+
+          <label htmlFor="speed">Speed: </label>
+          <select name="speed" value={values.speed} onChange={this.handleChange}>
+            <option value="undefined">default</option>
+            {Object.keys(constants.SpriteSpeed).map(type => {
+              return <option key={type} value={constants.SpriteSpeed[type]}>{type.replace(/_/g, ' ').toLowerCase()}</option>;
+            })}
+          </select>
+
+          <label htmlFor="size">Size: </label>
+          <select name="size" value={values.size} onChange={this.handleChange}>
+            <option value="undefined">default</option>
+            {Object.keys(constants.SpriteSize).map(type => {
+              return <option key={type} value={constants.SpriteSize[type]}>{type.replace(/_/g, ' ').toLowerCase()}</option>;
+            })}
+          </select>
+
+          <label htmlFor="direction">Direction: </label>
+          <select name="direction" value={values.direction} onChange={this.handleChange}>
+            <option value="undefined">default</option>
+            {Object.keys(constants.Direction).map(type => {
+              return <option key={type} value={constants.Direction[type]}>{type.replace(/_/g, ' ').toLowerCase()}</option>;
+            })}
+          </select>
+
+          <label htmlFor="emotion">Emotion: </label>
+          <select name="emotion" value={values.emotion} onChange={this.handleChange}>
+            <option value="undefined">default</option>
+            {Object.keys(constants.Emotions).map(type => {
+              return <option key={type} value={constants.Emotions[type]}>{type.replace(/_/g, ' ').toLowerCase()}</option>;
+            })}
+          </select>
+        </div>
+      );
+    }
+
+    return (
+      <form className="span4 offset1">
+        <header>
+          <strong>Editing Cell ({this.props.row}, {this.props.col})</strong>
+        </header>
+
+        <label htmlFor="tileType">Tile Type (required):</label>
+        <select name="tileType" value={values.tileType} onChange={this.handleChange}>
+          <option value={constants.SquareType.OPEN}>open</option>
+          <option value={constants.SquareType.SPRITEFINISH}>goal</option>
+          <option value={constants.SquareType.SPRITESTART}>sprite</option>
+        </select>
+
+        {spriteAttributes}
```

Comment:
An approach I've been taking is to do something like:

```
{(values.tileType === constants.SquareType.SPRITESTART) && 
<div> spriteAttributes jsx here </div>
}
```
Which allows you to have all of your jsx live in your family in the same structure it will render at.